sqlmesh/core/context.py

```diff
@@ -860,17 +860,18 @@ def format(
         self,
         transpile: t.Optional[str] = None,
         append_newline: t.Optional[bool] = None,
+        *,
+        check: t.Optional[bool] = None,
```

Comment:
Why do we need `*` here?

Reply:
I originally added it (in part) to help with the typing `@overload`s that I had (but later got the ok to remove). It's not techincally "needed" per se, but I think it's advisable. The `*` is a special parameter, which you can read about [here](https://docs.python.org/3/tutorial/controlflow.html#special-parameters). Essentially it ensures that the `check` argument is always used as a keyword argument and never a positional argument. This is helpful because something like `format(check=True)` is way more clear than just `format(True)`. Technically it would probably be even better for the `*` to be moved up 2 lines, but I was trying to avoid messing with your APIs too much.